src/ucp/wireup/ep_match.c

```diff
@@ -65,12 +65,13 @@ int ucp_ep_match_insert(ucp_worker_h worker, ucp_ep_h ep, uint64_t dest_uuid,
 {
     ucs_assert((conn_queue_type == UCS_CONN_MATCH_QUEUE_UNEXP) ||
                !(ep->flags & UCP_EP_FLAG_REMOTE_ID));
-    /* NOTE: protect union */
-    ucs_assert(!(ep->flags & (UCP_EP_FLAG_ON_MATCH_CTX |
-                              UCP_EP_FLAG_FLUSH_STATE_VALID)));
     /* EP matching is not used in CM flow */
     ucs_assert(!ucp_ep_has_cm_lane(ep));
 
+    /* NOTE: protect union */
+    ucs_assert(!(ep->flags & UCP_EP_FLAG_ON_MATCH_CTX));
+    ucp_ep_flush_state_invalidate(ep);
```

Comment:
IMO better ep_match.c will not change ep's flush_state, instead the caller of ep_match_insert/remove should do it

Reply:
actually, it aligns ep_match_insert/ep_match_retrieve with ep_match_remove and simplifies UCP EP code which don't need to call invalidate/reset when calling insert/retrieve.
also, IMO it is ok to have it here since `flush_state` shouldn't be filled when the EP is on E matching hash.

Reply:
ok